internal/serve/httphandler/tss_handler_test.go

```diff
@@ -0,0 +1,213 @@
+		}`, opXDRBase64)
+
+		rw := httptest.NewRecorder()
+		req := httptest.NewRequest(http.MethodPost, endpoint, strings.NewReader(reqBody))
+
+		mockRouter.
+			On("Route", mock.Anything).
+			Return(nil).
+			Once()
+
+		http.HandlerFunc(handler.SubmitTransactions).ServeHTTP(rw, req)
+		resp := rw.Result()
+		respBody, err := io.ReadAll(resp.Body)
+		require.NoError(t, err)
+
+		assert.Equal(t, http.StatusOK, resp.StatusCode)
+
+		var txSubmissionResp TransactionSubmissionResponse
+		_ = json.Unmarshal(respBody, &txSubmissionResp)
+
+		assert.Equal(t, 1, len(txSubmissionResp.TransactionHashes))
+
+		mockRouter.AssertNumberOfCalls(t, "Route", 1)
+	})
+}
+
+func TestGetTransaction(t *testing.T) {
+	dbt := dbtest.Open(t)
+	defer dbt.Close()
+
+	dbConnectionPool, err := db.OpenDBConnectionPool(dbt.DSN)
+	require.NoError(t, err)
+	defer dbConnectionPool.Close()
+	store, _ := store.NewStore(dbConnectionPool)
+	mockRouter := router.MockRouter{}
+	mockAppTracker := apptracker.MockAppTracker{}
+
+	handler := &TSSHandler{
+		Router:            &mockRouter,
+		Store:             store,
+		AppTracker:        &mockAppTracker,
+		NetworkPassphrase: "testnet passphrase",
+	}
+
+	endpoint := "/tss/transactions"
+
+	r := chi.NewRouter()
+	//r.Get("/tss/{transactionhash}", handler.GetTransaction)
+	r.Route(endpoint, func(r chi.Router) {
+		r.Get("/{transactionhash}", handler.GetTransaction)
+	})
+
+	clearTransactions := func(ctx context.Context) {
+		_, err = dbConnectionPool.ExecContext(ctx, "TRUNCATE tss_transactions")
+		require.NoError(t, err)
+	}
+
+	t.Run("returns_transaction", func(t *testing.T) {
+		txHash := "hash"
+		ctx := context.Background()
+		_ = store.UpsertTransaction(ctx, "localhost:8080/webhook", txHash, "xdr", tss.RPCTXStatus{OtherStatus: tss.NewStatus})
+		req, err := http.NewRequest(http.MethodGet, path.Join(endpoint, txHash), nil)
+		require.NoError(t, err)
+
+		// Serve request
+		rw := httptest.NewRecorder()
+		r.ServeHTTP(rw, req)
+		resp := rw.Result()
+		respBody, err := io.ReadAll(resp.Body)
+		require.NoError(t, err)
+		var getTxResp GetTransactionResponse
+		_ = json.Unmarshal(respBody, &getTxResp)
+
+		assert.Equal(t, http.StatusOK, resp.StatusCode)
+		assert.Equal(t, "hash", getTxResp.Hash)
+		assert.Equal(t, "xdr", getTxResp.XDR)
+		assert.Equal(t, "NEW", getTxResp.Status)
+
+		clearTransactions(ctx)
+	})
+
+	t.Run("return_empty_transaction", func(t *testing.T) {
+		txHash := "hash"
+		req, err := http.NewRequest(http.MethodGet, path.Join(endpoint, txHash), nil)
+		require.NoError(t, err)
+
+		// Serve request
+		rw := httptest.NewRecorder()
+		r.ServeHTTP(rw, req)
+		resp := rw.Result()
+		respBody, err := io.ReadAll(resp.Body)
+		require.NoError(t, err)
+		var getTxResp GetTransactionResponse
+		_ = json.Unmarshal(respBody, &getTxResp)
+
+		assert.Equal(t, http.StatusOK, resp.StatusCode)
+		assert.Empty(t, getTxResp.Hash)
+		assert.Empty(t, getTxResp.XDR)
+		assert.Empty(t, getTxResp.Status)
```

Comment:
I think we could return 404 instead of 200 when the transaction hash doesn't exist.

Reply:
Done